gantry/firmware/axis_x_hardware_config.c

```diff
@@ -0,0 +1,17 @@
+#include "axis_hardware_config.h"
+
+GantryHardwarePin gantry_hardware_get_gpio(GantryHardwareDevice device) {
+    GantryHardwarePin pinout;
+    switch(device) {
+        case gantry_hardware_device_LED_drive:
```

Comment:
This will fall through every time and always return GPIOB pin 7, needs a `break` statement in each case or a `return`. The compiler will then complain that there's a path to returning an uninitialized value, so you'll need a `default`.